xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAESink.cpp

```diff
@@ -728,6 +728,111 @@ void CActiveAESink::PrintSinks(std::string& driver)
   }
 }
 
+std::string CActiveAESink::ValidateOuputDevice(const std::string& device)
+{
+  if (m_sinkInfoList.empty())
+    return {};
+
+  const AESinkDevice dev = CAESinkFactory::ParseDevice(device);
+
+  // find exact match of deviceName in same driver
+  if (!dev.driver.empty() && !dev.name.empty())
+  {
+    for (const auto& sink : m_sinkInfoList)
+    {
+      if (sink.m_sinkName != dev.driver)
+        continue;
+
+      for (const auto& d : sink.m_deviceInfoList)
+      {
+        if (d.m_deviceName == dev.name)
+        {
+          std::string device = dev.driver + ":" + dev.name;
+
+          // note that friendly name may have changed, then is used
+          // current "physical" name (not name saved in settings)
+          // in this way friendly name is updated over time (drivers updates, etc.)
+          const std::string friendlyName = d.GetFriendlyName();
+          if (!friendlyName.empty())
+          {
+            device.append(":");
+            device.append(friendlyName);
+          }
+
+          return device;
+        }
+      }
+    }
+  }
+
+  // find same friendly name on other device in same driver
+  if (!dev.driver.empty() && !dev.friendlyName.empty())
+  {
+    for (const auto& sink : m_sinkInfoList)
+    {
+      if (sink.m_sinkName != dev.driver)
+        continue;
+
+      for (const auto& d : sink.m_deviceInfoList)
+      {
+        if (d.GetFriendlyName() == dev.friendlyName)
+          return dev.driver + ":" + d.m_deviceName + ":" + dev.friendlyName;
+      }
+    }
+  }
+
+  std::string firstDevice;
+
+  // find default device of same driver or first device of same driver
+  if (!dev.driver.empty())
+  {
+    for (const auto& sink : m_sinkInfoList)
+    {
+      if (sink.m_sinkName != dev.driver)
+        continue;
+
+      for (const auto& d : sink.m_deviceInfoList)
+      {
+        if (firstDevice.empty())
+        {
+          firstDevice = dev.driver + ":" + d.m_deviceName;
+
+          const std::string friendlyName = d.GetFriendlyName();
+          if (!friendlyName.empty())
+          {
+            firstDevice.append(":");
+            firstDevice.append(friendlyName);
+          }
+        }
+
+        if (d.m_deviceName.find("default") != std::string::npos)
+          return dev.driver + ":" + d.m_deviceName;
+      }
+
+      if (!firstDevice.empty())
+        break;
+    }
+  }
+
+  // return first device of same driver
+  if (!firstDevice.empty())
+    return firstDevice;
+
+  // fallback to first device of the list
+  firstDevice = m_sinkInfoList[0].m_sinkName;
```

Comment:
Please check that m_sinkInfoList and sinkInfoList[0].m_deviceInfoList are not empty first

Reply:
Done but only is need check `m_deviceInfoList` because `m_sinkInfoList` is already check at top of method